src/main/java/hudson/plugins/cigame/rules/plugins/findbugs/AbstractFindBugsWarningsRule.java

```diff
@@ -55,18 +64,33 @@ protected int getNumberOfAnnotations(List<FindBugsResultAction> list) {
     	}
     	
     	List<FindBugsResultAction> currentActions = ActionRetriever.getResult(build, Result.UNSTABLE, FindBugsResultAction.class);
-    	if (!hasNoErrors(currentActions)) {
-    		return EMPTY_RESULT;
-    	}
-    	int currentAnnotations = getNumberOfAnnotations(currentActions);
-    		
-    	List<FindBugsResultAction> previousActions = ActionRetriever.getResult(previousBuild, Result.UNSTABLE, FindBugsResultAction.class);
-    	if (!hasNoErrors(previousActions)) {
-    		return EMPTY_RESULT;
-    	}
-    	int previousAnnotations =getNumberOfAnnotations(previousActions);
-    	
-    	return evaluate(previousAnnotations, currentAnnotations);
+		if (currentActions.isEmpty()) {
+			return evaluateMaven(previousBuild, build);
+		} else {
+			if (!hasNoErrors(currentActions)) {
+				return EMPTY_RESULT;
+			}
+			int currentAnnotations = getNumberOfAnnotations(currentActions);
+			
+			List<FindBugsResultAction> previousActions = ActionRetriever.getResult(previousBuild, Result.UNSTABLE, FindBugsResultAction.class);
+			if (!hasNoErrors(previousActions)) {
+				return EMPTY_RESULT;
+			}
+			int previousAnnotations = getNumberOfAnnotations(previousActions);
+			
+			return evaluate(previousAnnotations, currentAnnotations);
+		}
+	}
+
+	private RuleResult<Integer> evaluateMaven(AbstractBuild<?, ?> previousBuild, AbstractBuild<?, ?> build) {
```

Comment:
is there no way to generify this method? this code is _very_ similar to the code in lines 73-81